src/main/java/com/zepben/cimbend/get/NetworkConsumerClient.kt

```diff
@@ -130,67 +149,74 @@ class NetworkConsumerClient(
             mRIDs.addAll(service.getUnresolvedReferenceMrids(Resolvers.location(it)))
         }
 
-        getIdentifiedObjects(service, mRIDs)
-            .onError { thrown, wasHandled -> return@getFeeder GrpcResult.ofError(thrown, wasHandled) }
-
-        return GrpcResult.of(feeder)
-    }
-
-    private fun processIdentifiedObjects(service: NetworkService, request: GetIdentifiedObjectsRequest): Sequence<IdentifiedObject?> {
-        return stub.getIdentifiedObjects(request)
-            .asSequence()
-            .map { it.objectGroup }
-            .flatMap {
-                sequenceOf(extractIdentifiedObject(service, it.identifiedObject)) +
-                    it.ownedIdentifiedObjectList.map { owned -> extractIdentifiedObject(service, owned) }
+        return GrpcResult.of(
+            getIdentifiedObjects(service, mRIDs).onError { thrown, wasHandled -> return@getFeeder GrpcResult.ofError(thrown, wasHandled) }.value.let {
+                MultiObjectResult(
+                    it.result.toMutableMap().apply { this[feeder.mRID] = feeder; this.putAll(equipmentResults.value.result) },
+                    it.failed.union(equipmentResults.value.failed))
             }
+        )
     }
 
-    private fun extractIdentifiedObject(service: NetworkService, it: NetworkIdentifiedObject): IdentifiedObject {
+    private fun processIdentifiedObjects(service: NetworkService, mRIDs: Iterable<String>): Sequence<ExtractResult> =
+        mRIDs.filter { service.get<IdentifiedObject>(it) == null }.let {  // Only process mRIDs not in the service
+            stub.getIdentifiedObjects(GetIdentifiedObjectsRequest.newBuilder().addAllMrids(mRIDs).build())
```

Comment:
Likewise this should not be passing in mRIDs to addAllMrids. It should be addAllMrids(it).